improver_tests/wxcode/wxcode/test_WeatherSymbols.py

```diff
@@ -500,52 +421,107 @@ def setUp(self):
         }
 
     def test_basic(self):
-        """Test create_condition_chain returns a list of strings."""
+        """Test create_condition_chain returns a nested list of iris.Constraint, 
+        floats, and strings representing operators that extracts the correct data."""
         plugin = WeatherSymbols()
         test_condition = self.dummy_queries["significant_precipitation"]
+        for t in test_condition["diagnostic_thresholds"]:
+            t.convert_units("m s-1")
+        thresholds = [t.points.item() for t in test_condition["diagnostic_thresholds"]]
         result = plugin.create_condition_chain(test_condition)
-        expected = (
-            "(cubes.extract(iris.Constraint(name='probability_of_"
-            "rainfall_rate_above_threshold', rainfall_rate=lambda "
-            "cell: 0.03 * {t_min} < "
-            "cell < 0.03 * {t_max}))[0].data >= 0.5) | (cubes.extract"
-            "(iris.Constraint("
-            "name='probability_of_lwe_snowfall_rate_above_threshold',"
-            " lwe_snowfall_rate=lambda cell: 0.03 * {t_min} < cell < "
-            "0.03 * {t_max}))[0].data >= 0.5)".format(
-                t_min=(1.0 - WeatherSymbols().float_tolerance),
-                t_max=(1.0 + WeatherSymbols().float_tolerance),
-            )
-        )
+        expected = [
+            [
+                [
+                    iris.Constraint(
+                        name="probability_of_rainfall_rate_above_threshold",
+                        rainfall_rate=lambda cell: np.isclose(
+                            cell.point,
+                            thresholds[0],
+                            rtol=WeatherSymbols().float_tolerance,
```

Comment:
[optional] There's a few of these `WeatherSymbols().float_tolerance` "floating" around. It makes sense to me to do `plugin.float_tolerance` instead. I.e. use the attribute from the instantiated plugin, as opposed to instantiating a new `WeatherSymbols` object just to get the tolerance float. However, I understand that's from the existing code. (Or maybe I'm missing something as to why it's written that way in the first place.)

Reply:
Fixed, it does make it a lot shorter to just use `plugin`